assets/js/ViewerApp.jsx

```diff
@@ -22,26 +24,38 @@ class ViewerApp extends Component {
       .join()
       .receive('ok', () => {});
 
-    channel.on('sign_update', ({ sign_id: signId, line_number: lineNumber, text: content }) => {
+    channel.on('sign_update', ({
+      sign_id: signId, duration, line_number: lineNumber, text: content,
+    }) => {
       this.setState(prevState => ({
-        signs: updateSigns(prevState.signs, { signId, lineNumber, content }),
+        signs: updateSigns(prevState.signs, {
+          signId, duration, lineNumber, content,
+        }),
       }));
     });
+
+    window.setInterval(this.updateTime, 1000);
+  }
+
+  updateTime() {
+    this.setState(({
+      currentTime: Date.now(),
+    }));
   }
 
   render() {
     const { signs } = this.state;
-
+    const { currentTime } = this.state;
```

Comment:
You can combine this with the above to be `const { signs, currentTime } = this.state;`